osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/domain/valueobjects/DomainType.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec60870.domain.valueobjects;
+
+public enum DomainType {
```

Comment:
I do partially agree, for most protocol adapters the response messages are also domain specific. For being able to send domain specific messages, some domain knowledge will be needed. Otherwise we would have to create either protocol specific response messages that will need to be understood by the domain adapters (introducing protocol knowledge in the domain adapters) or we would have to define such generic messages so that they can be understood by both protocol and domain adapters, or ???
The knowledge of domain was mainly introduced as a means for knowing to which domain adapter the responses should be routed to. For this routing I also added another story on the backlog (FLEX-5304, which also contains a still unanswered question about some routing options, maybe you can have a look as well?)
In the meantime it will probably be sufficient for now to reuse the domain info transferred from the request metadata into the response metadata, which should be available in the client connection event listener/asdu handlers...